File: scratch/extract_bi_dashboard.js
const fs = require('fs');

const lines = fs.readFileSync('middleware/src/routes/bi.js', 'utf8').split('\n');

lines.forEach((line, idx) => {
    if (line.includes('router.get') || line.includes('router.post') || line.includes('dashboard-data') || line.includes('executive-summary')) {
        console.log(`Line ${idx + 1}: ${line.trim()}`);
    }
});

File: scratch/extract_bi_dashboard.py
import re

with open('middleware/src/routes/bi.js', 'r', encoding='utf-8') as f:
    lines = f.readlines()

# Search for routes or endpoints
for idx, line in enumerate(lines):
    if 'router.get' in line or 'router.post' in line or 'dashboard-data' in line or 'executive-summary' in line:
        print(f"Line {idx+1}: {line.strip()}")
